engine/crates/engine/src/registry/mod.rs

```diff
@@ -1914,6 +1914,22 @@ impl Registry {
         }
     }
 
+    pub fn remove_empty_types(&mut self) {
+        let mut to_be_removed = Vec::new();
+
+        for r#type in self.types.values().filter(|ty| ty.is_object()) {
+            match r#type.fields() {
+                None => to_be_removed.push(r#type.name().to_owned()),
+                Some(fields) if fields.is_empty() => to_be_removed.push(r#type.name().to_owned()),
+                Some(_) => (),
+            }
+        }
+
+        for type_name in to_be_removed {
+            self.types.remove(&type_name);
+        }
+    }
```

Comment:
I wonder (though I'm not sure) whether this also needs to consider places that the removed types are used....

Reply:
I wondered about that. Currently, that would lead to a validation error right after the registry is generated. It's hard to say if it's going to be a problem in practice.

Reply:
I suppose this change is unlikely to make anything _more_ broken, so maybe we can go for it as is and improve if it's not good enough.